ADAL/tests/ADAuthenticationViewControllerTests.m

```diff
@@ -67,6 +67,39 @@ - (ADAuthenticationContext *)getTestAuthenticationContext
     return context;
 }
 
+
+- (void)testAboutBlankWhitelistInWebView
+{
+    //Create a context with empty token cache
+    ADAuthenticationContext* context = [self getTestAuthenticationContext];
+    
+    //Add two delegate calls to the mocking ADTestAuthenticationViewController
+    //First one in https while the second one in http
+    [ADTestAuthenticationViewController addDelegateCallWebAuthShouldStartLoadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:TEST_AUTHORITY]]];
+    [ADTestAuthenticationViewController addDelegateCallWebAuthShouldStartLoadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"about:blank"]]];
+    [ADTestAuthenticationViewController addDelegateCallWebAuthShouldStartLoadRequest:[NSURLRequest requestWithURL:TEST_REDIRECT_URL]];
```

Comment:
Just add a fake auth code at the end of this and feed ADTestURLConnection the proper token response.